packages/tendermint-rpc/src/client.ts

```diff
@@ -248,14 +249,43 @@ export class Client {
     };
   }
 
-  public async validators(height?: number): Promise<responses.ValidatorsResponse> {
+  public async validators(params: requests.ValidatorsParams): Promise<responses.ValidatorsResponse> {
     const query: requests.ValidatorsRequest = {
       method: requests.Method.Validators,
-      params: { height: height },
+      params: params,
     };
     return this.doCall(query, this.p.encodeValidators, this.r.decodeValidators);
   }
 
+  public async validatorsAll(params: requests.ValidatorsParams): Promise<responses.ValidatorsResponse> {
+    let page = params.page || 1;
+    const validators: responses.Validator[] = [];
+    let done = false;
+    let blockHeight = 0;
+
+    while (!done) {
+      const resp = await this.validators({
+        per_page: 50,
```

Comment:
Let's just process params into args before the loop, using the input as defaults.

```ts
let page = params.page || 1;
let per_page = params.per_page || 50;
let height = params.height || (await this.status()).height;
let args = {per_page, height}

//....
const resp = await this.validators({...args, page});
```

Reply:
`this.status()` doesn't return a height. I'll leave it out for this PR. Is it something we should be parsing?

Reply:
Sure, that or from the latest header.